static/js/settings_users.js

```diff
@@ -573,15 +573,31 @@ export function show_edit_user_info_modal(user_id, status_field) {
             profile_data: JSON.stringify(profile_data),
         };
 
-        settings_ui.do_settings_change(channel.patch, url, data, status_field);
-        dialog_widget.close_modal();
+        if (!from_user_info_popover) {
+            settings_ui.do_settings_change(channel.patch, url, data, status_field);
+            dialog_widget.close_modal();
+            return;
```

Comment:
Why does the other code path close the modal before `do_settings_change` returns, rather than having that be in a `success` handler?  I think that probably represents a bug / legacy settings feedback style in the existing flow that we should clean up.

Reply:
It is because the feedback for success/failure is shown in the settings overlay besides the `Users` heading.

Reply:
Yeah, I figured -- I think we probably want the feedback to be in the modal -- that's usually better UX since then you have the opportunity to fix issues if we get an error, etc.

Worth discussing a bit in #design but I think that's the pattern we're trying to move all of our modals to.